src/Symfony/Component/Console/Command/LockableTrait.php

```
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Command;
+
+use Symfony\Component\Console\Exception\RuntimeException;
+use Symfony\Component\Filesystem\LockHandler;
+
+/**
+ * Basic lock feature for commands.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+trait LockableTrait
+{
+    protected $lock;
+
+    /**
+     * Locks a command.
+     *
+     * @param string $name
+     * @param bool   $blocking
+     *
+     * @return bool
+     *
+     * @throws RuntimeException
+     */
+    protected function lock($name, $blocking = false)
+    {
+        if (!class_exists('Symfony\Component\Filesystem\LockHandler')) {
+            throw new RuntimeException('To enable the locking feature you must install the symfony/filesystem component');
+        }
+
+        $this->lock = new LockHandler($name);
```

If there is already a lock, we should probably throw an exception here.